src/net/sourceforge/kolmafia/persistence/ItemDatabase.java

```diff
@@ -2391,11 +2391,7 @@ public static boolean isAllowedInStandard(final int itemId) {
   }
 
   public static int getNoobSkillId(final int itemId) {
-    Integer skillId = ItemDatabase.noobSkillIdByItemSource.get(itemId);
-    if (skillId == null) {
-      return -1;
-    }
-    return skillId.intValue();
```

Comment:
Maybe I'm not digging deep enough but I cannot find a guarantee that ItemDatabase.noobSkillIdByItemSource.get(itemId) will never be null.  If it's not there ArrayList will throw an IndevOutOfBoundsException but null is AFAIK a perfectly reasonable element for an ArrayList.

Reply:
the return value of `get(itemId)` is also an `int`, so we just casting it to an Integer, so we were basically doing
```
int -> Integer -> null check
```

Reply:
this is because we use `IntegerArray`
```java
  public int get(final int index) {
    return index < 0 || index >= this.internalList.size() ? 0 : this.internalList.get(index);
  }
```

While it is true that `internalList.get(index)` could return `null`, it will produce a NullPointerException when trying to cast it to an int

Reply:
While I am not certain I agree with your analysis, I am convinced that noobSkillIdByItemSource will only contain ints and adding/setting a null would cause the add/set to error.  So, for different reasons, we agree that skillId will never be null.

However in the process of tracking this down I note that ItemFinder has code that processes a -1 returned from getNoobSkillId.  I'd love to see that code removed because getNoobSkillId will never return a -1 or have an example where it will return a -1.  But I won't mark this as unresolved.